packages/sessions/src/trackers/remote/index.ts

```diff
@@ -56,21 +56,36 @@ export class RemoteConfigTracker implements ConfigTracker, migrator.PresignedMig
     wallet: string
     digest: string
     chainId: ethers.BigNumberish
-    signatures: string[]
+    signatures: string[] | SignerSignature[]
   }): Promise<void> {
     let filteredSignatures = args.signatures
     if (this.onlyRecoverable) {
       filteredSignatures = filteredSignatures.filter(signature => {
-        return commons.signer.canRecover(signature)
-      })
+        if (typeof signature === 'string') {
+          return commons.signer.canRecover(signature)
+        } else {
+          // We "recover" using the included address
+          return !!signature.address
+        }
+      }) as string[] | SignerSignature[]
     }
 
-    await this.sessions.saveSignerSignatures({
-      wallet: args.wallet,
-      digest: args.digest,
-      chainID: numberString(args.chainId),
-      signatures: filteredSignatures
-    })
+    if (filteredSignatures.length === 0 || typeof args.signatures[0] === 'string') {
```

Comment:
Why not use `saveSignerSignatures2` every time? This check seems a bit buggy, as it assumes that the whole list of signatures will use the same type.

Reply:
I wanted to keep `saveWitnesses` backwards compatible. 
I'll update this to check that the entire list matches a format. To make this error happen they would have to ignore the type `string[] | SignerSignature[]` though. We generally accept that the types are as specified.

Reply:
what does it mean backwards compatible? like being able to roll this sequence.js update before sessions is updated?

Reply:
But couldn't you recover the address if `string[]` and then use `saveSignerSignatures2`?

Reply:
I don't think so? Correct me if I'm wrong but we'd need the wallet config version, which this function doesn't have access to. We could assume, but that seems even more error prone than using the current method as is.